lib/session.js

```diff
@@ -111,24 +154,23 @@ class Database {
 
   // list all sessions of a given type
   list({ type, sort, filter, limit }) {
-    let sessions = this.sessions.get(type, Map()).map(withSpeed);
+    let sessions, index;
     if (sort === 'count' || sort === 'per_hour') {
-      sessions = sessions.sort(
-        (a, b) => aggregateSpeed(b, 'per_hour') - aggregateSpeed(a, 'per_hour')
-      );
+      index = this.indexes.getIn([type, '24h'], Map());
     } else if (sort === 'speed' || sort === 'per_minute') {
-      sessions = sessions.sort(
-        (a, b) =>
-          aggregateSpeed(b, 'per_minute') - aggregateSpeed(a, 'per_minute')
-      );
+      index = this.indexes.getIn([type, '15m'], Map());
+    }
+    index = index.sort().reverse();
+    if (limit) {
```

Comment:
Shouldn't this happen after the `if (filter)` otherwise if asking for 100 results with only the `reputation: bad` we will receive less than 100 results as we will filter the not `bad` afterwards.